fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1156,6 +1173,29 @@ public Builder setIndex(@Nonnull String indexName) {
             return this;
         }
 
+        public Builder setTargetIndexes(@Nonnull List<Index> indexes) {
+            this.targetIndexes = new ArrayList<>(indexes);
+            return this;
+        }
+
+        public Builder setTargetIndexesByNames(@Nonnull List<String> indexes) {
+            final RecordMetaData metaData = getRecordMetaData();
+            return setTargetIndexes(indexes.stream().map(metaData::getIndex).collect(Collectors.toList()));
+        }
+
+        public Builder addTargetIndex(@Nonnull Index index) {
```

Comment:
I think the relationship between `.setIndex` and the target index methods could use some explaining...or maybe it's a sign that there's something weird with how this API is shaping up.

In particular, it's a bit strange that you can set both the "index" and the "target index(es)" here. Possibly, if we deprecated the old `.setIndex` methods, then this could be made a little cleaner, but I'm not certain.